IoT-Sample/Swift/IoTSampleSwift/ConnectionViewController.swift

```diff
@@ -228,32 +294,66 @@ class ConnectionViewController: UIViewController, UITextViewDelegate {
         tabBarViewController.viewControllers = [ self, configurationViewController ]
         logTextView.resignFirstResponder()
 
-        // Initialize AWSMobileClient for authorization
-        AWSMobileClient.sharedInstance().initialize { (userState, error) in
-            guard error == nil else {
-                print("Failed to initialize AWSMobileClient. Error: \(error!.localizedDescription)")
-                return
-            }
-            print("AWSMobileClient initialized.")
-        }
+        //Initialize control plane
+        // Initialize the Amazon Cognito credentials provider
+        let credentialsProvider = AWSCognitoCredentialsProvider(regionType:AWS_REGION,
+                                                                identityPoolId:IDENTITY_POOL_ID)
+        let controlPlaneServiceConfiguration = AWSServiceConfiguration(region:AWS_REGION, credentialsProvider:credentialsProvider)
+
+        //IoT control plane seem to operate on iot.<region>.amazonaws.com
+        //Set the defaultServiceConfiguration so that when we call AWSIoTManager.default(), it will get picked up
+        AWSServiceManager.default().defaultServiceConfiguration = controlPlaneServiceConfiguration
+        iotManager = AWSIoTManager.default()
+        iot = AWSIoT.default()
+
         
-        // Init IOT
+        //Initialize Dataplane:
+        // IoT Dataplane must use your account specific IoT endpoint
         let iotEndPoint = AWSEndpoint(urlString: IOT_ENDPOINT)
         
-        // Configuration for AWSIoT control plane APIs
-        let iotConfiguration = AWSServiceConfiguration(region: AWSRegion, credentialsProvider: AWSMobileClient.sharedInstance())
-        
         // Configuration for AWSIoT data plane APIs
-        let iotDataConfiguration = AWSServiceConfiguration(region: AWSRegion,
+        let iotDataConfiguration = AWSServiceConfiguration(region: AWS_REGION,
                                                            endpoint: iotEndPoint,
-                                                           credentialsProvider: AWSMobileClient.sharedInstance())
-        AWSServiceManager.default().defaultServiceConfiguration = iotConfiguration
+                                                           credentialsProvider: credentialsProvider)
+        //IoTData manager operates on xxxxxxx-iot.<region>.amazonaws.com
+        AWSIoTDataManager.register(with: iotDataConfiguration!, forKey: AWS_IOT_DATA_MANAGER_KEY)
+        iotDataManager = AWSIoTDataManager(forKey: AWS_IOT_DATA_MANAGER_KEY)
+    }
 
-        iotManager = AWSIoTManager.default()
-        iot = AWSIoT.default()
+    func setupViewComponents() {
+        let labelHeight = 44 * 4
```

Comment:
nit: Why not set these up in the storyboard? It doesn't seem like code we'd care for customers to use as reference, so it's OK to make it on copy/paste friendly. The only interesting dynamic pieces look to be the labels on the button, which you could set up on viewDidLoad.

Reply:
Storyboards were super frustrating and as couldn't get things aligned... at all -- so i basically nuked everything in the storyboard and just went manual.